### PR TITLE
feat: distinct agent colors via golden-angle HSV

### LINKED ISSUE #25: Agents share duplicate colors — palette too small for population size
## Problem

Multiple agents can have the same color. The color palette has 8 fixed entries, but default population is 10 (max 100). Colors are assigned via `AGENT_COLORS[agent_id % 8]`, so agents 8 and 9 immediately duplicate agents 0 and 1. The `next_agent_id` counter also resets to 0 on every new generation, reproducing the same collision pattern.

### Palette (`agent/mod.rs:83-100`)

8 hardcoded sRGB triples: red, blue, green, yellow, purple, orange, cyan, pink.

```rust
pub fn agent_color(index: usize) -> [f32; 3] {
    AGENT_COLORS[index % AGENT_COLORS.len()]  // modulo-8 wrap
}
```

### Assignment (`agent/mod.rs:169`)

Color is set once in `Agent::new`:
```rust
color: agent_color(id as usize),
```

Where `id` is `App.next_agent_id`, a `u32` counter that resets to 0 on:
- Full reset (`main.rs:659`)
- `spawn_evolution_population` (`main.rs:675`)
- `spawn_population_from_configs` (`main.rs:727`)

### Population vs palette

| Constant | Value | Location |
|---|---|---|
| Palette size | 8 | `agent/mod.rs:83` |
| Default population | 10 | `xagent-shared/src/config.rs:223` |
| Max agents (`MAX_AGENTS`) | 100 | `agent/mod.rs:103` |
| UI slider range | 2–100 | `ui.rs:1443` |

At default settings, 2 color collisions. At max, 92 collisions.

### Color persistence

`reset_for_new_life` (`agent/mod.rs:256-274`) does NOT touch `self.color` — color is sticky per `Agent` struct. Dead agents render as `DEAD_COLOR = [0.3, 0.3, 0.3]` but retain their stored color for when alive.

## What to change

Replace the fixed palette with procedural generation using **golden-angle hue stepping**, which produces visually distinct colors for any population size without needing to know the count upfront:

```rust
pub fn agent_color(index: usize) -> [f32; 3] {
    let hue = (index as f32 * 137.508) % 360.0;  // golden angle
    let saturation = 0.75;
    let value = 0.85;
    hsv_to_srgb(hue, saturation, value)
}
```

This requires adding an `hsv_to_srgb` helper in `agent/mod.rs`. The `AGENT_COLORS` array and `DEAD_COLOR` constant can be removed (dead color is already handled separately at render time in `main.rs:1667-1673`).

### Files to modify

- **`crates/xagent-sandbox/src/agent/mod.rs`**:
  - Remove `AGENT_COLORS` array (lines 83-96)
  - Replace `agent_color` function (line 98) with golden-angle HSV generation
  - Add `hsv_to_srgb` helper
  - `DEAD_COLOR` can stay or be inlined at its single usage site

### Consumers (no changes needed — they all read `agent.color`)

- `main.rs:1667-1684` — instance buffer: reads `a.color`, applies `srgb_to_linear`
- `main.rs:1891` — snapshot bridge: copies `a.color` to `AgentSnapshot`
- `ui.rs:1103-1126` — minimap dots: reads `agent.color`
- `agent/mod.rs:570` — `generate_all_agents_mesh`: reads `agent.color`

### Tests

Verify no existing tests assert specific color values from the palette. Add a test that generates 100 colors and asserts all are unique (comparing with a small epsilon for float equality).

### PR DESCRIPTION
**Pull request by koraytaylan**
## Summary
- Replace the fixed 8-color palette in `agent_color()` with golden-angle HSV hue stepping (`hue = index * 137.508° mod 360°`, saturation 0.8, value 0.9)
- Every agent now gets a deterministic, visually distinct color regardless of population size — no more duplicate colors when `agent_count > 8`
- Added `hsv_to_rgb` helper for the conversion

## Test plan
- [x] `cargo check -p xagent-sandbox` passes
- [x] All 99 tests pass (`cargo test -p xagent-sandbox`)
- [ ] Visual inspection: run simulation with 20+ agents, verify no duplicate-looking colors

Closes #25

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/xagent-sandbox/src/agent/mod.rs [hunk 1]
```diff
@@ -76,27 +76,38 @@ impl AgentBody {
...
+        0 => (c, x, 0.0),
+        1 => (x, c, 0.0),
+        2 => (0.0, c, x),
+        3 => (0.0, x, c),
+        4 => (x, 0.0, c),
+        _ => (c, 0.0, x),
+    };
+    let m = v - c;
+    [r1 + m, g1 + m, b1 + m]
+}
```
[the remark this answers is not in the document]
Comment (koraytaylan) [reply crates/xagent-sandbox/src/agent/mod.rs]: Added `h.rem_euclid(360.0)` and `s/v.clamp(0.0, 1.0)` at the top of `hsv_to_rgb`. Also added a test (`hsv_to_rgb_normalizes_inputs`) that verifies negative hue wrap, hue > 360 wrap, and out-of-range s/v clamping.

File: crates/xagent-sandbox/src/agent/mod.rs [hunk 2]
```diff
@@ -76,27 +76,38 @@ impl AgentBody {
...
+
+/// Deterministic, visually distinct color for the agent at `index`.
+/// Uses golden-angle hue stepping so any population size gets well-separated colors.
 pub fn agent_color(index: usize) -> [f32; 3] {
-    AGENT_COLORS[index % AGENT_COLORS.len()]
+    let hue = (index as f32 * GOLDEN_ANGLE_DEG) % 360.0;
+    hsv_to_rgb(hue, 0.8, 0.9)
 }
```
[the remark this answers is not in the document]
Comment (koraytaylan) [reply crates/xagent-sandbox/src/agent/mod.rs:116]: Added three tests in the existing `mod tests` block:
- `hsv_to_rgb_outputs_in_unit_range` — asserts RGB in 0..=1 for hues 0, 30, 60, 120, 180, 240, 300, 359.999
- `hsv_to_rgb_normalizes_inputs` — verifies negative/overflow hue wrap and s/v clamping
- `first_16_agent_colors_are_distinct` — checks all 16 colors differ beyond epsilon
